shell/platform/android/io/flutter/view/FlutterNativeView.java

```diff
@@ -67,32 +72,30 @@ public long get() {
     }
 
     public void assertAttached() {
-        if (!isAttached())
-            throw new AssertionError("Platform view is not attached");
+        if (!isAttached()) throw new AssertionError("Platform view is not attached");
     }
 
-    public void runFromBundle(String bundlePath, String snapshotOverride, String entrypoint, boolean reuseRuntimeController) {
-        assertAttached();
-        if (applicationIsRunning)
-            throw new AssertionError("This Flutter engine instance is already running an application");
-
-        nativeRunBundleAndSnapshot(mNativePlatformView, bundlePath, snapshotOverride, entrypoint, reuseRuntimeController, mContext.getResources().getAssets());
-
-        applicationIsRunning = true;
+    public void runFromBundle(String bundlePath, String snapshotOverride, String entrypoint,
+            boolean reuseRuntimeController) {
+        runFromBundle(bundlePath, snapshotOverride, entrypoint, null, reuseRuntimeController, null);
     }
 
-    public void runFromSource(final String assetsDirectory, final String main, final String packages) {
+    public void runFromBundle(String bundlePath, String snapshotOverride, String entrypoint,
+            String libraryUrl, boolean reuseRuntimeController, FlutterIsolateStartedEvent event) {
```

Comment:
We should create an object that holds the parameters needed to launch the Dart runtime instead of adding more arguments here.  Some of the arguments of runFromBundle() added from past experiments are already obsolete (snapshotOverride, reuseRuntimeController).  I'd like to make it easier to evolve this interface.

@chinmaygarde had a proposal for refactoring the host APIs along these lines

Reply:
I think that's a good idea. Would you happen to know if there's a doc for this proposal?

Reply:
I've gone ahead and added `FlutterRunArguments` to hold all the parameters required for entering an isolate in a new shell. Added deprecation message for all other `runFromBundle` definitions that don't use this new argument object.